sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -52,16 +52,21 @@ case class CollectLimitExec(limit: Int, child: SparkPlan) extends LimitExec {
     SQLShuffleReadMetricsReporter.createShuffleReadMetrics(sparkContext)
   override lazy val metrics = readMetrics ++ writeMetrics
   protected override def doExecute(): RDD[InternalRow] = {
-    val locallyLimited = child.execute().mapPartitionsInternal(_.take(limit))
-    val shuffled = new ShuffledRowRDD(
-      ShuffleExchangeExec.prepareShuffleDependency(
-        locallyLimited,
-        child.output,
-        SinglePartition,
-        serializer,
-        writeMetrics),
-      readMetrics)
-    shuffled.mapPartitionsInternal(_.take(limit))
+    val childRDD = child.execute()
+    val singlePartitionRDD = if (childRDD.getNumPartitions != 1) {
```

Comment:
nit: could you leave some comments about why we cannot strip the shuffle if  `childRDD.getNumPartitions == 0`?

Reply:
refer to [CoalesceExec.doExecute](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala#L715), I update the fix to directly return empty RDD with single partition if `childRDD.getNumPartitions == 0`